### ISSUE TITLE
Iu-xray

### ISSUE THREAD

**Issue by maocheno** [mk-runner/MLRG#10]
Hello, thank you very much for your excellent work. I would like to know what your model’s CE metric is on IU‑Xray. I implemented your model myself and obtained P = 0.243, R = 0.249, F1 = 0.241. Does this match your results? If not, could you please provide your IU‑Xray metrics? Thank you!

**Comment by mk-runner** [comment]
Thank you very much for your interest in our work.

We did not conduct experiments directly on the IU X-ray dataset, as it does not contain longitudinal data. The dataset that is somewhat related is the [Two-view CXR dataset](https://huggingface.co/datasets/MK-runner/Multi-view-CXR) [1], which includes data from the IU X-ray dataset.

If you would like to compute the performance specifically on the IU X-ray subset, you can extract the relevant samples from the following file: `MLRG/generated-radiology-reports/Two-view CXR/test_reports_epoch-0_25-10-2024_11-38-35.csv`.

After isolating the IU X-ray data, you may proceed to calculate the NLG and CE metrics accordingly.

Please feel free to reach out if you have any further questions.

[1] EVOKE: Elevating Chest X-ray Report Generation via Multi-View Contrastive Learning and Patient-Specific Knowledge. 

Best regards,
Kang Liu

**Comment by maocheno** [comment]
Dear Dr. Liu,

Thank you very much for your kind reply. Your work is truly impressive, and I have learned a great deal from reading it.

I did notice that your released predictions include results on the IU X-ray dataset. However, we observed that IU X-ray was used during training in your current setup. Our primary focus is to evaluate zero-shot transfer performance—that is, performance on IU X-ray without any training on it.

The numbers we obtained in our evaluation were based on this zero-shot setting. To avoid any misunderstanding when citing your results in our paper, I wonder if it would be possible for you to kindly share the direct transfer performance of your model on IU X-ray, without any fine-tuning or training on that dataset?

This would be incredibly helpful for ensuring clarity and fairness in our comparison.

Thank you again for your time, and please let me know if there's anything I can provide from our side.

Best regards

**Comment by mk-runner** [comment]
Thank you again for your continued interest.

Our model was trained on the MIMIC-CXR dataset and directly applied to the Two-view CXR dataset for *zero-shot report generation*. Specifically:

* For samples originating from MIMIC-CXR, this is not strictly zero-shot.
* For samples derived from IU X-ray (included in Two-view CXR), the model has not seen them during training, so these cases are evaluated in a zero-shot setting.

You can find the full results of this experiment under:
**`MLRG/generated-radiology-reports/“zero-shot” report generation on the Two-view CXR Dataset`**

To evaluate the model performance *specifically* on IU X-ray samples, we extracted the relevant subset (total: 1223 samples) using the `dicom_ids`. 

We modified the following script for this evaluation:
**`script/MIMIC-CXR/run_cxr_test_mlrg_v1011.sh`**
by replacing the `ann_path` with the Two-view CXR annotation file (i.e., [hugging face](https://huggingface.co/MK-runner/MLRG/blob/main/radiology%20report/mlrg_multiview_cxr_annotation_v1.1.json)). `test_ckpt_path` is `https://huggingface.co/MK-runner/MLRG/blob/main/mimic-cxr/finetune/best_model.ckpt`

The performance metrics on IU X-ray (zero-shot setting) are as follows:

```
# Sample count: 1223
{
  'F1-Radgraph-partial': 0.2991,
  'chexbert_5_micro_f1': 0.3163,
  'chexbert_5_macro_f1': 0.3338,
  'chexbert_all_micro_p': 0.5645,
  'chexbert_all_micro_r': 0.5923,
  'chexbert_all_micro_f1': 0.5781,
  'chexbert_all_macro_p': 0.3645,
  'chexbert_all_macro_r': 0.2118,
  'chexbert_all_macro_f1': 0.2281,
  'BLEU_1': 0.3979,
  'BLEU_2': 0.2278,
  'BLEU_3': 0.1426,
  'BLEU_4': 0.0934,
  'METEOR': 0.1601,
  'ROUGE_L': 0.3027,
  'CIDEr': 0.1653
}
```

**Hope this helps! A star would be much appreciated.**

Please let us know if you need any further clarification.

Best regards,
Kang Liu
